gcn_email_service/helpers.py

```diff
@@ -0,0 +1,21 @@
+import threading
+
+
+def periodic_task(interval, times=-1):
+    def outer_wrap(function):
+        def wrap(*args, **kwargs):
+            stop = threading.Event()
+
+            def inner_wrap():
+                i = 0
+                while i != times and not stop.is_set():
+                    stop.wait(interval)
+                    function(*args, **kwargs)
+                    i += 1
```

Comment:
Remove support for `times`, and also catch any exception other than `KeyboardInterrupt` so that a failed evaluation does not exit from the loop:

```suggestion
                while not stop.is_set():
                    stop.wait(interval)
                    try:
                        function(*args, **kwargs)
                    except Exception:
                        logger.exception('Periodic task failed')
```